### PR TITLE
separate patches into dedicated source files

### PR DESCRIPTION
**Pull request by FranciscoDA**

### CODE REVIEW

File: patches/random_costs.cpp [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+bool isNaturalResourceCost(const ResourceCost &cost) {
+    return cost.Type != -1 && cost.Type < 4 && cost.Amount > 0;
+}
+
+bool isNaturalResearchResourceCost(const ResearchResourceCost &cost) {
+    return cost.Type != -1 && cost.Type < 4 && cost.Amount > 0;
+}
+
+bool hasNaturalResourceCost(const genie::Unit &unit) {
+	std::vector<ResourceCost> costs = unit.Creatable.ResourceCosts;
```
Comment (HSZemi) [inline patches/random_costs.cpp:21]: This looks like there are a few issues with spaces and tabs in this file :thinking:

File: CMakeLists.txt [hunk 2]
```diff
@@ -4,6 +4,19 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -static-libstdc++")
 add_subdirectory(genieutils EXCLUDE_FROM_ALL)
 include_directories(
         "genieutils/include"
+		"."
```
Comment (simonsan) [inline CMakeLists.txt:7]:
```suggestion
        "."
```

**Review by simonsan — APPROVED** [review]

File: patches/random_costs.cpp (hunk 1 above)
Comment (FranciscoDA) [reply patches/random_costs.cpp]: whoops, my bad

**Comment by simonsan** [comment]
Look good, really like that change, much better overview and easier to add new mods in the future! 👍🏽

**Comment by simonsan** [comment]
@FranciscoDA What did you use for code formatting, do you have a `.clang-format` you may want to commit to keep the code style. Seems clean and nice.

**Comment by FranciscoDA** [comment]
> @FranciscoDA What did you use for code formatting, do you have a `.clang-format` you may want to commit to keep the code style. Seems clean and nice.

Manually formatted I'm afraid :cold_sweat:  (although I honored the original formatting a bit)

I've added a .clang-format file and reformatted the sources according to that.

**Comment by HSZemi** [comment]
Thank you! :yellow_heart:

**Comment by simonsan** [comment]
Nice! 👍🏽

**Pull request merged**